### PR TITLE
Fix(eos_designs)!: BGP vlan config should not have redistribute igmp when belonging to a VRF with evpn multicast

### LINKED ISSUE #4061: Bug(eos_designs): BGP vlan config should not have `redistribute igmp` when belonging to a VRF with `evpn multicast`
- having `redistribute igmp` for a svi/vlan belonging to a VRF with `evpn multicast` is an issue because it could cause conflicts in the control-plane as far I know.

_Originally posted by @krikoon73 in https://github.com/aristanetworks/avd/issues/3462#issuecomment-1884638318_

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Having redistribute igmp for a svi/vlan belonging to a VRF with evpn multicast is an issue because it causes extra control-plane/data-plane load. https://github.com/aristanetworks/avd/issues/3462#issuecomment-1884638318

## Related Issue(s)

Fixes #4061 

## Component(s) name

`arista.avd.eos_designs`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py [hunk 1]
```diff
@@ -457,7 +457,7 @@ def _router_bgp_vlans_vlan(self: AvdStructuredConfigNetworkServices, vlan, tenan
             bgp_vlan["route_targets"]["import_export_evpn_domains"] = [{"domain": "remote", "route_target": vlan_rt}]
 
         vlan_evpn_l2_multicast_enabled = default(get(vlan, "evpn_l2_multicast.enabled"), get(tenant, "evpn_l2_multicast.enabled"))
-        if vlan_evpn_l2_multicast_enabled is True:
+        if vlan_evpn_l2_multicast_enabled is True and not vrf:
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py]: I think,
if vlan_evpn_l2_multicast_enabled is True and not self.network_services_l3:

is more correct.

Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py]: Please add a comment stating the reason why vrf should not be present.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Blocking to discuss if this should wait for 5.0, since it could be considered a breaking change.

**Comment by ClausHolbechArista** [comment]
Moving to draft until 4.10 has been merged.

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -75,6 +75,10 @@ The following ansible plugins have been removed from the `arista.avd` Ansible co
 
 TODO: Level 3 sections for each change with details on how to migrate
 
+### Redistribute IGMP
+
+Removing redistribute igmp configuration for a BGP svi/vlan belonging to a VRF with EVPN multicast.
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: we can add the diff here

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]: added

**Review by Shivani-gslab — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 3]
```diff
@@ -75,6 +75,26 @@ The following ansible plugins have been removed from the `arista.avd` Ansible co
 
 TODO: Level 3 sections for each change with details on how to migrate
 
+### Removed `redistribute igmp` from BGP vlan config belonging to a VRF with evpn multicast
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:78]: This is not really a porting guide entry. We need to describe the change (in words or a config snip. not a structured config snip) and how to revert the change if you prefer to.
So we need to show how the user can keep the redistribute igmp. If we do not have a knob for that, we should add that.

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 4]
```diff
@@ -155,6 +155,12 @@ With AVD version 5.0.0 `ipv6_address_virtuals` are only rendered in the configur
 
 The `eos_designs.jsonschema.json` is no longer generated. This schema was not being used and had never been complete.
 
+#### Removed redistribute igmp from BGP vlan config belonging to a VRF with evpn multicast
+
+Having redistribute igmp configured for a BGP svi/vlan belonging to a VRF with EVPN multicast causes conflicts in the control-plane.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]: This description needs to be improved. I will ask someone for help on this.

**Review by ClausHolbechArista — COMMENTED** [review]
Please add a knob in the datamodel if we don't have it, to allow the user to configure the redistribute igmp per vlan.

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md (hunk 3 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/docs/porting-guides/5.x.x.md:78]: Added a knob `redistribute_igmp_vrf`

**Review by ClausHolbechArista — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM!
